### PR TITLE
Feature damage

### PR DESCRIPTION
**Pull request by Iambananayeah**
## Description

merge damage system into dev

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by jelly-konjac — APPROVED** [review]

File: Silksong/Assets/Scripts/MapObjects/AttackTest/Bullet.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        transform.Translate(transform.right*Time.deltaTime*facingRight*speed, Space.World);
+        if(transform.position.x<-20 || transform.position.x>20)//��Ҫ�޸�
```
Comment (IsaacJade) [inline Silksong/Assets/Scripts/MapObjects/AttackTest/Bullet.cs:23]: 这个之后可以在镜头左右两边视角看不到的地方，放几个透明hitbox，碰到就消除

Comment (Iambananayeah) [reply Silksong/Assets/Scripts/MapObjects/AttackTest/Bullet.cs:23]: 好的 暂时测试就先这样了

File: Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+/// <summary>
+/// 
+/// ��ʱ������ҵ���ͨ��������� ���������
+/// </summary>���ߣ����
+public class PlayerAttack : MonoBehaviour
```
Comment (IsaacJade) [inline Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs:8]: 不用coroutine等机制的话，不用继承mono吧，数据之后可以一起写一个配置类。mono多了管理也混乱

File: Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+   // private PlayerController PlayerController;
+    void Start()
+    {
+        //PlayerController = GetComponent<PlayerController>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if(Input.GetKeyDown(KeyCode.C))//������
```
Comment (IsaacJade) [inline Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs:26]: 建议封装成单独方法，方便之后接入fsm。以及使用封装好的input system获取按键信息

File: Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            else
+            {
+                bltPostion.x += 0.5f;
+                blt.GetComponent<Bullet>().facingRight = 1;
+            }
+            blt.transform.position = bltPostion;
+        }
+
+
+        if (Input.GetKeyDown(KeyCode.X) && isAttacking==false)//��ͨ����
```
Comment (IsaacJade) [inline Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs:45]: 同上

File: Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs (hunk 2 above)
Comment (Iambananayeah) [reply Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs:8]: 好的

File: Silksong/Assets/Scripts/MapObjects/Collision/Collider/ColliderBase.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+/// <summary>
+/// ��ײ���ĳ����� ��һ����ʽ��Ҫ
+/// </summary>���ߣ����
+public abstract class ColliderBase : MonoBehaviour
```
Comment (IsaacJade) [inline Silksong/Assets/Scripts/MapObjects/Collision/Collider/ColliderBase.cs:7]: 不建议在此次pull中更新碰撞体相关

File: Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs (hunk 3 above)
Comment (Iambananayeah) [reply Silksong/Assets/Scripts/MapObjects/AttackTest/PlayerAttack.cs:26]: 攻击这方面都是临时写来测试的，对接了再写正式的

**Review by IsaacJade — COMMENTED** [review]
Pass

**Review by IsaacJade — APPROVED** [review]

**Pull request merged**